src/lib.rs

```diff
@@ -42,8 +42,13 @@ pub async fn run_trin(
     tokio::spawn(Arc::clone(&discovery).bucket_refresh_lookup());
 
     // Initialize metrics
-    if trin_config.enable_metrics {
-        let binding = trin_config.metrics_url.as_ref().unwrap().parse().unwrap();
+    if trin_config.enable_metrics_with_url.is_some() {
+        let binding = trin_config
+            .enable_metrics_with_url
+            .as_ref()
+            .unwrap()
+            .parse()
+            .unwrap();
```

Comment:
It would be nice to handle these unwraps and display a proper error message if the user inputs an invalid address:port